Technology Bonuses WIP

kubiix opened the pull request:
Work in progress on Technolgy page - part II not mentioned to finish te PR

- extended tooltip for technolgy, not sure if put there also unlocked recipes, now only units count and ingredients
- added two lists to technology page for unlocked items and unlocked fluids
- [Experimental] added list with technology bonuses, like braking force, ammo damage, since the bonuses are not prototype based, it is not simple to get needed info to display correct all possible bonuses from base game, not mentioning possible new bonuses from mods like QoL research (I included it to PR just to show you if you are interested in it, but as i see it now, it would cost too much work with quite small benefit to the mod, and it should be removed)

raiguard commented:
Thanks for your hard work! I do like the bonuses and you implemented them well, but I don't think they're really in-scope for a Recipe Book. Additionally, we can't use the actual bonus icons, so I agree that it should be removed.

The items and fluids and tooltip are great though! Thank you. Just one small change: could you move the item and fluid lists up above the prerequisites list? Perhaps put them in alphabetical order (fluid, item recipe).

kubiix commented:
I have removed all the Bonus related code and reordered lists on technology page.
Also refactored formatter and extracted tooltip formatters into separate files (hope it is not too cheeky from me)

raiguard commented:
Thanks, I'll go ahead and merge it and do some other formatter cleanup as well.
